sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1203,14 +1283,19 @@ private void flushBatch() throws DatastoreException, IOException, InterruptedExc
           CommitRequest.Builder commitRequest = CommitRequest.newBuilder();
           commitRequest.addAllMutations(mutations);
           commitRequest.setMode(CommitRequest.Mode.NON_TRANSACTIONAL);
+          long startTime = System.currentTimeMillis();
           datastore.commit(commitRequest.build());
+          long endTime = System.currentTimeMillis();
+
+          writeBatcher.addRequestLatency(endTime, endTime - startTime, mutations.size());
+
           // Break if the commit threw no exception.
           break;
         } catch (DatastoreException exception) {
           // Only log the code and message for potentially-transient errors. The entire exception
```

Comment:
Ideally. Not for all errors though, of course. I added taking into account latency for DEADLINE_EXCEEDED errors only, which seems like the reasonable approach.